### DISCUSSION TITLE
Use this to build custom IDE

### DISCUSSION THREAD

**Discussion by ReIaxo** [dotnet/project-system#9404] [Q&A]
Hi, 

we work on an IDE-ish project and need to "handle" dotnet project files. We support the old .csproj format already but not the new SDK-style.
Is this the right project to go with for supporting SDK-based projects? Or is it only applicable for Visual Studio extensions?
We need to show a project tree like Visual Studio does.

I do not really understand the [Feature Comparison](https://github.com/dotnet/project-system/blob/main/docs/feature-comparison.md) regarding ASP.NET: In which cases is it supported?

Thank you in advance.

**Comment by drewnoakes** [comment]
@Relaxo this repo is built on top of an internal component (CPS &mdash; Common Project System) that's not usable outside of Visual Studio. You can use the code in this repo to understand what kinds of integrations you might need to support when implementing .NET project support in your project.

**Comment by lextm** [comment]
If you want to build a custom IDE based on code from this repo, you can use it along with [my CPS shims](https://github.com/lextudio/cps-shims).
